DIPs/DIP1337.md

```diff
@@ -0,0 +1,101 @@
+# Tail Call Optimization requirement
+
+| Section         | Value                                                           |
+|-----------------|-----------------------------------------------------------------|
+| DIP:            | 1000                                                            |
+| Author:         | Dietrich Daroch (name dot surname on gmail)                     |
+| Implementation: | (links to implementation PR if any)                             |
+| Status:         | -                                                               |
+
+
+
+## Abstract
+
+Add a way to enforce tail call optimization.
+
+
+### Links
+
+[NASA's Jet Propulsion Lab C guidelines](http://lars-lab.jpl.nasa.gov/JPL_Coding_Standard_C.pdf) ban recursion because of the uncertainty.
+
+
```

Comment:
you mentioned `@tailrec` from Scala, that's probably a good example for your motivation